### PR TITLE
react-input: Updated to use new shared tokens

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated input styles to use shared tokens from react-theme

## Issues

Updates #22231

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d325a31bcd5d144e2a40ee4bf085e8ef6b3f5d73'>d325a31bcd5d144e2a40ee4bf085e8ef6b3f5d73</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/Avatar-mount_base.html">933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/Avatar-mount.html">953</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/Button-mount_base.html">596</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/Button-mount.html">589</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProvider-mount_base.html">1992</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProvider-mount.html">1935</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-mount_base.html">264</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-mount.html">285</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">245</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">258</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">307</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">323</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/MakeStyles-mount_base.html">1563</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22836/perf-test-react-components/MakeStyles-mount.html">1630</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-input/src/components/Input/useInputStyles.ts [hunk 1]
```diff
@@ -15,32 +15,9 @@ export const inputClassNames: SlotClassNames<InputSlots> = {
 };
 
 // TODO(sharing) use theme values once available
```
Comment (spmonahan) [inline packages/react-components/react-input/src/components/Input/useInputStyles.ts:17]: Can we remove this comment now?

File: packages/react-components/react-input/src/components/Input/useInputStyles.ts [hunk 2]
```diff
@@ -15,32 +15,9 @@ export const inputClassNames: SlotClassNames<InputSlots> = {
...
-  // TODO(sharing) shouldn't these be in the theme?
-  body1: {
-    fontSize: tokens.fontSizeBase300,
-    lineHeight: tokens.lineHeightBase300,
-  },
-  caption1: {
-    fontSize: tokens.fontSizeBase200,
-    lineHeight: tokens.lineHeightBase200,
-  },
+  // TODO: This 400 style isnot in the typography styles.
```
Comment (spmonahan) [inline packages/react-components/react-input/src/components/Input/useInputStyles.ts:19]:
```suggestion
  // TODO: This 400 style is not in the typography styles.
```

**Review by spmonahan — APPROVED** [review]

File: packages/react-components/react-input/src/components/Input/useInputStyles.ts (hunk 1 above)
Comment (GeoffCoxMSFT) [reply packages/react-components/react-input/src/components/Input/useInputStyles.ts:17]: There is the leftover contentSizes[400] so I left that comment there.

**Pull request merged**